catalyst_voices/apps/voices/lib/widgets/document_builder/single_line_https_url_widget.dart.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:catalyst_voices/widgets/text_field/voices_https_text_field.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_view_models/catalyst_voices_view_models.dart';
+import 'package:flutter/material.dart';
+
+class SingleLineHttpsUrlWidget extends StatefulWidget {
+  final DocumentProperty<String> property;
+  final String description;
+  final bool isEditMode;
+  final ValueChanged<DocumentChange> onChanged;
+  const SingleLineHttpsUrlWidget({
+    super.key,
+    required this.property,
+    required this.description,
+    required this.isEditMode,
+    required this.onChanged,
+  });
+
+  @override
+  State<SingleLineHttpsUrlWidget> createState() =>
+      _SingleLineHttpsUrlWidgetState();
+}
+
+class _SingleLineHttpsUrlWidgetState extends State<SingleLineHttpsUrlWidget> {
+  late TextEditingController _textEditingController;
+  late final FocusNode _focusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    _textEditingController = TextEditingController();
+    final value = widget.property.value;
+    if (value != null) {
+      _textEditingController.text = value;
+    }
+    _textEditingController.addListener(_handleControllerChange);
+    _focusNode = FocusNode(canRequestFocus: widget.isEditMode);
+  }
+
+  @override
+  void didUpdateWidget(covariant SingleLineHttpsUrlWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (oldWidget.isEditMode != widget.isEditMode &&
+        widget.isEditMode == false) {
+      _textEditingController.text = widget.property.value ?? '';
+    }
+
+    if (widget.isEditMode != oldWidget.isEditMode) {
+      _handleEditModeChanged();
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _textEditingController.dispose();
+  }
```

Comment:
super.dispose() goes always last. This is from the docs of `dispose()` method:
```
  /// Implementations of this method should end with a call to the inherited
  /// method, as in `super.dispose()`.
```


```suggestion
  @override
  void dispose() {
      _textEditingController.dispose();
    super.dispose();
  }
```